### ISSUE TITLE
crash when connecting my usb key

### ISSUE THREAD

**Issue by greg0ire** [coldfix/udiskie#96]
I have a crash when I connect a USB key I just formatted.

```
** (udiskie:4558): WARNING **: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-fjfqFxeBfH: Connexion refusée
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/udiskie/dbus.py", line 263, in __call__
    return self._handler(object_path, *parameters.unpack())
  File "/usr/local/lib/python2.7/dist-packages/udiskie/udisks2.py", line 812, in _job_completed
    self.trigger('job_failed', device, action, message)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/udisks2.py", line 657, in trigger
    super(Daemon, self).trigger(event, device, *args)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/common.py", line 51, in trigger
    handler(*args)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/notify.py", line 166, in job_failed
    text = _('failed to {0} {1}:\n{2}', action, device_file, message)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/locale.py", line 41, in __call__
    return msg.format(*args, **kwargs)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 687: ordinal not in range(128)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/udiskie/dbus.py", line 70, in _callback
    self.errback(e)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 78, in errback
    if not self._finish(self.errbacks, exception):
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 69, in _finish
    fn(*args)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 262, in _throw
    self._interact(self._generator.throw, exc)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 275, in _interact
    self.errback(e)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 78, in errback
    if not self._finish(self.errbacks, exception):
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 69, in _finish
    fn(*args)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 262, in _throw
    self._interact(self._generator.throw, exc)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 275, in _interact
    self.errback(e)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 78, in errback
    if not self._finish(self.errbacks, exception):
  File "/usr/local/lib/python2.7/dist-packages/udiskie/async_.py", line 69, in _finish
    fn(*args)
  File "/usr/local/lib/python2.7/dist-packages/udiskie/mount.py", line 114, in _error
    fn.__name__, device, err.message))
  File "/usr/local/lib/python2.7/dist-packages/udiskie/locale.py", line 41, in __call__
    return msg.format(*args, **kwargs)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 792: ordinal not in range(128)
```

I am french, and this kind of error often happens when `é` appears somewhere in the data.

**Comment by coldfix** [comment]
Hey, thanks for the info!

Can you describe in more detail what's your setup:

- udiskie version
- python2 or python3
- udisks1 or udisks2
- is the special character in the volume name or somewhere else?
- what kind of filesystem is on the device

**Comment by coldfix** [comment]
Okay, I believe I have fixed one issue with unicode strings on python2. This should have been a long standing issue, I wonder why only now someone reported it.

Can you checkout the branch `fix-unicode-problem` and try if it fixes your problem?

```
pip install git+git://github.com/coldfix/udiskie@fix-unicode-problem
```

**Comment by greg0ire** [comment]
I tried with the version you recommended, without success (I get the exact same errors).
udiskie reports version `1.4.2`, is that what I should have if I install things correctly ?

```
udiskie --version        
udiskie 1.4.2
/usr/bin/python --version
Python 2.7.9
dpkg -l|grep udisks
ii  gir1.2-udisks-2.0:amd64                                     2.1.5-1                                     amd64        GObject based library to access udisks2 - introspection data
ii  libudisks2-0:amd64                                          2.1.5-1                                     amd64        GObject based library to access udisks2
ii  udisks                                                      1.0.5-1build1                               amd64        storage media interface
ii  udisks2
```

I do not know where the special character is…

```
ls -l /dev/disk/by-label
total 0
lrwxrwxrwx 1 root root 10 déc.  23 10:37 ARCH_201512 -> ../../sdg1
lrwxrwxrwx 1 root root 10 déc.  23 10:28 Racine -> ../../sdf1
```

The disk was formatted in NTFS, like this : 

`sudo mkfs.ntfs -f /dev/sdg1`

**Comment by greg0ire** [comment]
I tried formatting with a label with no special characters, and it works, so I guess this really is the problem.

**Comment by greg0ire** [comment]
Well that's strange, it works with "Grégoire"…

**Comment by coldfix** [comment]
> Well that's strange, it works with "Grégoire"… is it possible that a caching system or something like that made the use of the new version not instant ? I think I changed terminals between tries…

It could be... Did you remove your udiskie system installation before installing the new version?

Can you confirm that the problem is fixed with the bugfix branch? Otherwise please post the current stack trace (just in case the problem has changed and you didn't notice).

**Comment by greg0ire** [comment]
Sorry, I won't be able to test that soon. I did not remove the previous installation before installing the new one. I guess I should try again without any label at all.

**Comment by coldfix** [comment]
Ok, no problem. I have uploaded the bugfix branch along with other changes as a new version by now. I hope those will fix your problem. Otherwise please report back to me;)

**Comment by coldfix** [comment]
This problem should be fixed now. Report back, if you experience further problems.
